linkis-computation-governance/linkis-engineconn/linkis-engineconn-plugin-core/src/main/scala/org/apache/linkis/manager/engineplugin/common/util/NodeResourceUtils.scala

```diff
@@ -35,18 +40,35 @@ object NodeResourceUtils {
         properties.get(EngineConnPluginConf.JAVA_ENGINE_REQUEST_MEMORY.key)
       if (StringUtils.isBlank(settingClientMemory)) {
         properties.remove(EngineConnPluginConf.JAVA_ENGINE_REQUEST_MEMORY.key)
-      } else if (
-          !settingClientMemory.toLowerCase().endsWith("g")
-          && !settingClientMemory.toLowerCase().endsWith("m")
-      ) {
+      } else {
         properties.put(
           EngineConnPluginConf.JAVA_ENGINE_REQUEST_MEMORY.key,
-          settingClientMemory + "g"
+          formatJavaOptionMemoryWithDefaultUnitG(settingClientMemory)
         )
       }
     }
   }
 
+  def formatJavaOptionMemory(memory: String, defaultUnit: String): String = {
+    assert(StringUtils.isNotBlank(memory))
+    val memoryTmp = memory.toUpperCase(Locale.getDefault)
+    if (
+        memoryTmp.endsWith(JAVA_MEMORY_UNIT_G) || memoryTmp.endsWith(
+          JAVA_MEMORY_UNIT_M
+        ) || memoryTmp.endsWith(JAVA_MEMORY_UNIT_K)
+    ) {
+      memoryTmp
+    } else if (memoryTmp.endsWith(JAVA_MEMORY_UNIT_B)) {
```

Comment:
Xmx not support  unit MB,
If the value unit set by the user  not within the range（G/M/K/B）, whether to consider print  warn log and set it as the default memory 1g

Reply:
This is a compatiable with exchangis, if use GB, MB, KB, B,  just drop it.  is this ok?

Reply:
This is okay. If the user accidentally enters the wrong unit such as GK, do we need to consider using the default value or add some warn logs

Reply:
in this case,  the check should use regex to check?   if the unit wit `GK`, the ec will startup failed.

Reply:
> in this case, the check should use regex to check? if the unit wit `GK`, the ec will startup failed.

I think it's ok